binr/ragg2/ragg2.c

```diff
@@ -108,6 +123,420 @@ static int openfile (const char *f, int x) {
 }
 #define ISEXEC (fmt!='r')
 
+static char* getCompiler(void) {
+	char *output = NULL;
+	char *CC = NULL;
+
+	output = r_sys_getenv ("CC");
+	// if CC is not set, or CC is not gcc, llvm-gcc, or clang
+	if (!output || (strcmp (output, "llvm-gcc") 
+  		&& strcmp (output, "clang") && strcmp (output, "gcc")))
+	{
+	   	output = r_sys_cmd_strf ("which llvm-gcc");
+		if (output) {
+            if (!(CC = strdup ("llvm-gcc"))) {
+				goto fail;
+			}
+		}
+
+		free (output);
+		output = r_sys_cmd_strf ("which clang");
+		if (output) {
+			if (!(CC = strdup ("clang"))) {
+				goto fail;
+			}
+		}
+
+		free (output);
+		output = r_sys_cmd_strf ("which gcc");
+		if (output) {
```

Comment:
make CC const char * and remove all those strdups/free/ifs

Reply:
and gotos

Reply:
If I make CC const char, there is not point to call r_sys_get_env anymore, do we agree ? I will juste check what compiler is isntaller with the 3 tests (which clang, which gcc ,...)